src/test_snmp.cpp

```diff
@@ -86,6 +88,190 @@ TEST_F(SNMPTest, LatencyCalculations)
   delete tbl;
 }
 
+TEST_F(SNMPTest, LatencyScopeCalculations)
+{
+  cwtest_completely_control_time(true);
+
+  // Create a table.  Use a different OID base because otherwise the indices
+  // clash with later tests.
+  SNMP::EventAccumulatorByScopeTable* tbl = SNMP::EventAccumulatorByScopeTable::create("latency", test_oid);
+
+  // Just put one sample in (which should have a variance of 0).  This should
+  // show up in only the current 5 minute stats.
+  tbl->accumulate(100);
+
+  // Check the counts
+  ASSERT_EQ(0, snmp_get(".1.2.2.1.7.1.4.110.111.100.101"));
+  ASSERT_EQ(1, snmp_get(".1.2.2.1.7.2.4.110.111.100.101")); // Current 5 minutes
+  ASSERT_EQ(0, snmp_get(".1.2.2.1.7.3.4.110.111.100.101"));
+
+  // Move on five seconds. The "previous five seconds" stat should now reflect that sample.
+  cwtest_advance_time_ms(5000);
+
+  // Average should be 100
+  ASSERT_EQ(100, snmp_get(".1.2.2.1.3.1.4.110.111.100.101"));
+  // Variance should be 0
+  ASSERT_EQ(0, snmp_get(".1.2.2.1.4.1.4.110.111.100.101"));
+
+  // Now input two samples in this latency period.
+  tbl->accumulate(300);
+  tbl->accumulate(500);
+
+  // Move on five seconds. The "previous five seconds" stat should now reflect those two samples.
+  cwtest_advance_time_ms(5000);
+
+  // Average should be 400
+  ASSERT_EQ(400, snmp_get(".1.2.2.1.3.1.4.110.111.100.101"));
+  // HWM should be 500
+  ASSERT_EQ(500, snmp_get(".1.2.2.1.5.1.4.110.111.100.101"));
+  // LWM should be 300
+  ASSERT_EQ(300, snmp_get(".1.2.2.1.6.1.4.110.111.100.101"));
+  // Count should be 2
+  ASSERT_EQ(2, snmp_get(".1.2.2.1.7.1.4.110.111.100.101"));
+
+  // Move on 10 minutes.  Everything should be zero again.
+  cwtest_advance_time_ms(1000*60*10);
+
+  for (int time_period : {1, 2, 3})
+  {
+    for (int stat : {3, 4, 5, 6, 7})
+    {
+      ASSERT_EQ(0, snmp_get(".1.2.2.1." + std::to_string(stat) + "." + std::to_string(time_period) + ".4.110.111.100.101"));
+    }
+  }
+
+  cwtest_reset_time();
+  delete tbl;
+}
+
+enum EventStats
+{
+  MEAN = 1,
+  VAR = 2,
+  HWM = 3,
+  LWM = 4,
+  COUNT = 5
+};
+
+enum TimeScope
+{
+  PREV_5S = 1,
+  CURR_5M = 2,
+  PREV_5M = 3
+};
+
+TEST_F(SNMPTest, StringBasedEventStats)
+{
+  cwtest_completely_control_time(true);
+
+  // Create a table.  Use a different OID base because otherwise the indices?// clash with later tests.
```

Comment:
Feel like this is a typo here. Also, as above, you're using the same `test_oid`.